application/tests/org.openjdk.jmc.jolokia.test/src/test/java/org/openjdk/jmc/jolokia/JolokiaTest.java

```diff
@@ -0,0 +1,163 @@
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+ * WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.openjdk.jmc.jolokia;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import javax.management.Attribute;
+import javax.management.AttributeNotFoundException;
+import javax.management.InstanceNotFoundException;
+import javax.management.IntrospectionException;
+import javax.management.InvalidAttributeValueException;
+import javax.management.MBeanAttributeInfo;
+import javax.management.MBeanException;
+import javax.management.MBeanServerConnection;
+import javax.management.MalformedObjectNameException;
+import javax.management.ObjectName;
+import javax.management.OperationsException;
+import javax.management.ReflectionException;
+import javax.management.remote.JMXConnector;
+import javax.management.remote.JMXServiceURL;
+
+import org.awaitility.Awaitility;
+import org.eclipse.core.runtime.preferences.InstanceScope;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.openjdk.jmc.common.IDescribable;
+import org.openjdk.jmc.jolokia.preferences.PreferenceConstants;
+import org.openjdk.jmc.rjmx.IConnectionDescriptor;
+import org.openjdk.jmc.rjmx.IServerDescriptor;
+import org.openjdk.jmc.rjmx.descriptorprovider.IDescriptorListener;
+
+/**
+ * I test that JMX connections done with JmcJolokiaJmxConnectionProvider are functional 
+ */
+@SuppressWarnings("restriction")
+public class JolokiaTest {
+
+	static String jolokiaUrl;
+
+	private static Set<String> unsafeAttributes = new HashSet<>(
+			Arrays.asList("BootClassPath", "UsageThreshold", "UsageThresholdExceeded", "UsageThresholdCount",
+					"CollectionUsageThreshold", "CollectionUsageThresholdExceeded", "CollectionUsageThresholdCount"));
+
+	private static JolokiaDiscoveryListener discoveryListener;
+
+	private static MBeanServerConnection jolokiaConnection;
+
+	@BeforeClass
+	public static void startServer() throws Exception {
+		// wait for Jolokia to be ready before commencing tests
+		Awaitility.await().atMost(Duration.ofSeconds(15))//Note: hard code property to avoid module dependency on agent
+				.until(() -> (jolokiaUrl = System.getProperty("jolokia.agent")) != null);
+		discoveryListener = new JolokiaDiscoveryListener();
+		jolokiaConnection=getJolokiaMBeanConnector();
+		
+
+	}
+
+	@Test
+	public void testReadAttributesOverJolokia() throws MalformedURLException, IOException, OperationsException,
+			IntrospectionException, AttributeNotFoundException, ReflectionException, MBeanException {
+		for (ObjectName objectName : jolokiaConnection.queryNames(null, null)) {
```

Comment:
Iterating over all attributes may require some maintenance. However it will also detect any problems with this connector for new JVM developments. 
I can consider just testing a few select attributes instead.